### PR TITLE
feat: add the ability to share a link to a log line

### PR DESCRIPTION
**Pull request by dnazarenkoo**

### CODE REVIEW

File: frontend/src/container/LogsExplorerList/index.tsx [hunk 1]
```diff
@@ -61,6 +79,28 @@ function LogsExplorerList({
 		[options],
 	);
 
+	const handleCopyLogLink = useCallback(
+		(id: string) => {
+			const timeRange = JSON.stringify(copiedTimeRange);
+			const params = new URLSearchParams(window.location.search);
+
+			params.delete(LogLinkQueryParams.activeLogId);
```
Comment (palashgdev) [inline frontend/src/container/LogsExplorerList/index.tsx:87]: use this is one `src/constants/query.ts`

File: frontend/src/container/LogsExplorerList/index.tsx [hunk 2]
```diff
@@ -61,6 +79,28 @@ function LogsExplorerList({
...
+			params.delete(LogLinkQueryParams.timeRange);
+			params.set(LogLinkQueryParams.activeLogId, `"${id}"`);
+			params.set(LogLinkQueryParams.timeRange, timeRange);
+
+			const link = `${window.location.origin}${
```
Comment (palashgdev) [inline frontend/src/container/LogsExplorerList/index.tsx:92]: use Location hook

File: frontend/src/container/LogsExplorerList/index.tsx [hunk 3]
```diff
@@ -61,6 +79,28 @@ function LogsExplorerList({
 		[options],
 	);
 
+	const handleCopyLogLink = useCallback(
```
Comment (palashgdev) [inline frontend/src/container/LogsExplorerList/index.tsx:82]: can we make a hook for above logic and use i form there

File: frontend/src/types/api/index.ts [hunk 4]
```diff
@@ -7,9 +7,10 @@ export interface ErrorResponse {
 	message: null;
 }
 
-export interface SuccessResponse<T> {
+export interface SuccessResponse<T, P = unknown> {
 	statusCode: SuccessStatusCode;
 	message: string;
 	payload: T;
 	error: null;
+	params?: P;
```
Comment (palashgdev) [inline frontend/src/types/api/index.ts:15]: why we are making change this one ?

File: frontend/src/hooks/queryBuilder/useGetExplorerQueryRange.ts [hunk 5]
```diff
@@ -47,6 +48,7 @@ export const useGetExplorerQueryRange = (
 			selectedTime: 'GLOBAL_TIME',
 			globalSelectedInterval,
 			query: requestData || initialQueriesMap.metrics,
+			params,
```
Comment (palashgdev) [inline frontend/src/hooks/queryBuilder/useGetExplorerQueryRange.ts:51]: is options not working ?

File: frontend/src/container/LogsExplorerViews/index.tsx [hunk 6]
```diff
@@ -168,8 +179,27 @@ function LogsExplorerViews(): JSX.Element {
...
+				}),
+		},
 	);
 
+	useEffect(() => {
+		const params = data?.params as Omit<CopiedTimeRange, 'timeRange'>;
+
+		if (!params || logs.length <= 0) return;
+
+		setCopiedTimeRange((prevData) => ({
```
Comment (palashgdev) [inline frontend/src/container/LogsExplorerViews/index.tsx:196]: this should be handled when we copy not in useEffect

File: frontend/src/container/LogsExplorerViews/index.tsx [hunk 7]
```diff
@@ -84,6 +92,9 @@ function LogsExplorerViews(): JSX.Element {
 	const [page, setPage] = useState<number>(1);
 	const [logs, setLogs] = useState<ILog[]>([]);
 	const [requestData, setRequestData] = useState<Query | null>(null);
+	const [copiedTimeRange, setCopiedTimeRange] = useState<CopiedTimeRange>(
```
Comment (palashgdev) [inline frontend/src/container/LogsExplorerViews/index.tsx:95]: no use of this state

File: frontend/src/container/LogsExplorerList/index.tsx (hunk 2 above)
Comment (dnazarenkoo) [reply frontend/src/container/LogsExplorerList/index.tsx:92]: I can't use origin from useLocation(

File: frontend/src/hooks/queryBuilder/useGetExplorerQueryRange.ts (hunk 5 above)
Comment (dnazarenkoo) [reply frontend/src/hooks/queryBuilder/useGetExplorerQueryRange.ts:51]: not really, I'm using it for fetch API params, not query hook params

File: frontend/src/container/LogsExplorerViews/index.tsx (hunk 7 above)
Comment (dnazarenkoo) [reply frontend/src/container/LogsExplorerViews/index.tsx:95]: do you mean avoiding this state?

File: frontend/src/api/metrics/getQueryRange.ts [hunk 8]
```diff
@@ -18,6 +18,7 @@ export const getMetricsQueryRange = async (
 			error: null,
 			message: response.data.status,
 			payload: response.data,
+			params: props,
```
Comment (palashgdev) [inline frontend/src/api/metrics/getQueryRange.ts:21]: can you please confirm api accept params also ?

File: frontend/src/container/LogsExplorerList/constants.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export enum LogLinkQueryParams {
```
Comment (palashgdev) [inline frontend/src/container/LogsExplorerList/constants.ts:1]: no need for this constant as global QueryParam is already present

File: frontend/src/container/LogsExplorerViews/index.tsx (hunk 7 above)
Comment (palashgdev) [reply frontend/src/container/LogsExplorerViews/index.tsx:95]: yes this is handled here 

```
https://github.com/SigNoz/signoz/blob/e97609ce23b95da97cfe362a43e976884ebb17f4/frontend/src/container/TopNav/DateTimeSelection/index.tsx#L37
```

File: frontend/src/types/api/index.ts (hunk 4 above)
Comment (palashgdev) [reply frontend/src/types/api/index.ts:15]: you can make use of hook for same and utilise options

Comment (dnazarenkoo) [reply frontend/src/types/api/index.ts:15]: it just returns our input params (body), why do we have to create a hook for that?

**Review by nityanandagohain — CHANGES_REQUESTED** [review]
Change highlight color to light orange

**Review by nityanandagohain — CHANGES_REQUESTED** [review]
* In Showing 10 lines after match, the `id` to be sent should be the last log id and not the first one.

**Review by nityanandagohain — APPROVED** [review]

**Comment by palashgdev** [comment]
Closed in favor of https://github.com/SigNoz/signoz/pull/3190

**Pull request closed without merging**